--- include/boost/detail/endian.hpp
+++ include/boost/detail/endian.hpp
@@ -64,8 +64,8 @@
    || defined(_M_ALPHA) || defined(__amd64) \
    || defined(__amd64__) || defined(_M_AMD64) \
    || defined(__x86_64) || defined(__x86_64__) \
-   || defined(_M_X64) || defined(__bfin__)
-
+   || defined(_M_X64) || defined(__bfin__) \
+   || defined(ANDROID)
 # define BOOST_LITTLE_ENDIAN
 # define BOOST_BYTE_ORDER 1234
 #else
