diff --git a/project-config.jam b/project-config.jam
index 271ceb1..0b45250 100644
--- a/project-config.jam
+++ b/project-config.jam
@@ -21,11 +21,12 @@ using python : 2.6 : /usr ;
 # options. If left empty, all libraries will be built.
 # Options specified on the command line completely
 # override this variable.
-libraries =  ;
+# @Moss - Adding needed libs to be valid for android
+libraries = --with-date_time --with-filesystem --with-program_options --with-regex --with-signals --with-system --with-thread --with-iostreams ; 
 
 # These settings are equivivalent to corresponding command-line
 # options.
-option.set prefix : /usr/local ;
-option.set exec-prefix : /usr/local ;
-option.set libdir : /usr/local/lib ;
-option.set includedir : /usr/local/include ;
+option.set prefix : ./../build/ ;
+option.set exec-prefix : ./../build/ ;
+option.set libdir : ./../build/lib ;
+option.set includedir : ./../build/include ;
